### PR TITLE
Added the source file name as data-source-file

### PR DESCRIPTION
**Pull request by TrevorFSmith**
If there are React.Component namespace conflicts then they can be more specifically selected using the original source filename.

### CODE REVIEW

File: __tests__/index-test.js [hunk 1]
```diff
@@ -36,6 +37,7 @@ class componentName extends Component {
 export default componentName;
 `,
     {
+      filename: "./filename-test.js",
```
Comment (patrick-fs) [inline __tests__/index-test.js:40]: To be sure: is this an attribute that is set in Babel config or can it be inferred during the compilation phase by the plugin visitor?

**Comment by patrick-fs** [comment]
Hey Trevor,

I see a `ClassDeclaration` that adds the `data-component` and `data-source-file` attributes to a React component class.

What about function components?

Take this example from our reference app: https://github.com/fullstorydev/react-shoppe-demo/blob/master/src/components/Products.tsx#L15

`<ProductGrid >` is a function component. Reading through the plugin code,  it looks like it will have a `data-element` attribute to it rather than a `data-component` and that it wouldn't have a `data-source-file` at all.

What are your thoughts?

**Comment by TrevorFSmith** [comment]
> What about function components?

Yes, I believe the naked function components aren't decorated because the current logic in the Babel plugin doesn't know which functions are being used as React.Components. I'm not sure whether there's enough state in Babel to determine that but I'll take a look.

**Comment by TrevorFSmith** [comment]
It looks like the logic in the `mn` version (the origin of the logic for this plugin) doesn't handle naked functions (see [removed test here](https://github.com/cowpaths/mn/pull/22720/files#diff-91f418405d62c9ac808a720bfab6b348)) but the latest data-qa plugin does ([test here](https://github.com/davesnx/babel-plugin-transform-react-qa-classes/blob/master/test/fixtures/react/option-format/expected.js)). So, I'll dig in and see what's different between the two. But, it seems fixable at least.

**Comment by TrevorFSmith** [comment]
The way it works is that when a React.Component is defined (be it via class or function) the root element of the rendered code has a `data-component` attribute. Both root and children have `data-element` attributes.

So, the rendered output would be something like this:

    <div data-component=App data-element=View>
        
        <div data-component=PizzaTranslator data-element=View>
            <input data-element=TextInput ... />
        </div>
    </div>

**Pull request merged**